test/Test_SpotifyAPI.cpp

```diff
@@ -350,8 +350,60 @@ TEST_F(SpotifyAPITest, playTryValidSong) {
       }
     } else {
       // device found, set volume on it
+
+      auto playRes = mAPI.play(
+          mAccessToken, vector<string>{"spotify:track:7HbGgElHEx4fpyzHYtfiox"});
+      if (playRes.has_value()) {
+        // if we reach here not possible
+        EXPECT_EQ(0, 1);
+      } else {
+        EXPECT_EQ(playRes, nullopt);
+      }
+    }
+  }
+}
+
+TEST_F(SpotifyAPITest, play2Times) {
+  // check if there are devices available
+  auto devicesRet = mAPI.getAvailableDevices(mAccessToken);
+  if (auto value = get_if<Error>(&devicesRet)) {
+    value = value;
+    cout << "Error happend in getting devices.. can't happen!!" << endl;
+    EXPECT_EQ(0, 1);
+  } else {
+    auto devices = get<vector<Device>>(devicesRet);
+    if (devices.empty()) {
+      // if devices are empty test set on no device
       auto playRes = mAPI.play(
           mAccessToken, vector<string>{"spotify:track:7HbGgElHEx4fpyzHYtfiox"});
+      if (!playRes.has_value()) {
+        // cant reach here
+        EXPECT_EQ(0, 1);
+      } else {
+        EXPECT_EQ(ErrorCode::SpotifyNotFound, playRes.value().getErrorCode());
+        EXPECT_EQ("Player command failed: No active device found",
+                  playRes.value().getErrorMessage());
+      }
+    } else {
+      // device found, set volume on it
+      auto playRes = mAPI.play(
+          mAccessToken, vector<string>{"spotify:track:7HbGgElHEx4fpyzHYtfiox"});
+      if (playRes.has_value()) {
+        // if we reach here not possible
+        EXPECT_EQ(0, 1);
+      } else {
+        EXPECT_EQ(playRes, nullopt);
+      }
+      sleep(2);
+      playRes = mAPI.play(mAccessToken);
+      if (playRes.has_value()) {
+        // if we reach here not possible
+        EXPECT_EQ(0, 1);
```

Comment:
you could actually just call `.value()` and take the exception if it is empty. GTEST will handle that